### ISSUE TITLE
Boxed enum types are treated as regular numbers

### ISSUE THREAD

**Issue by TriEdgeAI** [bridgedotnet/Bridge#2065]
### Expected

```
Boat
Boat
Boat
```

### Actual

```
Boat
2
2
```

### Steps To Reproduce

[Deck](http://deck.net/50c6936a439e6c4dcafdb53edfce9dee)

```cs
public enum VehicleType
{
    Car,
    Plane,
    Boat
}

public class Program
{
    public static void Main()
    {
        VehicleType vehicleType = VehicleType.Boat;
        object box = vehicleType;
        
        Console.WriteLine(vehicleType);
        Console.WriteLine(box);
        Console.WriteLine(Enum.Parse(typeof(VehicleType), "Boat"));
    }
}
```

**Comment by michaelcheers** [comment]
Related to #1347, #1317, #1312.

**Comment by geoffreymcgill** [comment]
Boxing/Unboxing support has been implemented, but moving release to the 16.0 milestone.

**Comment by michaelcheers** [comment]
Duplicate of [this comment](https://github.com/bridgedotnet/Bridge/issues/1309#issuecomment-217610825).
